src/modules/mysql.nix

```diff
@@ -120,20 +121,16 @@ in
       cfg.package
     ];
 
-    env.MYSQLDATA = config.env.DEVENV_STATE + "/mysql";
-    env.MYSQLSOCKET = config.env.DEVENV_STATE + "/mysql.sock";
+    mysql.settings.mysql.user = mkDefault "root";
+    mysql.settings.client.user = mkDefault config.mysql.settings.mysql.user;
+    mysql.settings.mysqldump.user = mkDefault config.mysql.settings.mysql.user;
```

Comment:
This sets the user that these tools will use by default.

The tools internally still use `-u root`, as this has the correct rights to setup databases and do ping.